### DISCUSSION TITLE
Help with setQueryData

### DISCUSSION THREAD

**Discussion by Ttislevoll** [TanStack/query#6850] [Q&A]
Hi!
I have some troubles with mutations. I have a code that loops trough a ordersQuery and run mutations on affected orders.

First, it finds the specified order to mutate:

```
const order = ordersQuery.data?.find(( --> Finds specific order))

orderMutation.mutate(produce(order, (draft) => {
              if (draft.weight== null) {
                draft.weight= 0;
              }
}
```

Then, it loops trough the orderQuery again, finds a list of affected orders and mutates these orders:

```
ordersQuery.data?.filter( --> Some filtering)).forEach((o) => {

orderMutation.mutate(produce(o, (draft) => {
            draft.affected= 1;
          }));
}
```

Finally, i loop trough the orders again, to ADD some weight

```
const order = ordersQuery.data?.find(( --> Finds specific order))

orderMutation.mutate(produce(order, (draft) => {
              // Some logic to set the desired newWeight//
              if (draft.weight== null) {
                draft.weight= newWeight;
              }
}
```

The issue is that when I get to mutation to ADD the property weight, the order from ordersQuery will have the un-mutated old values, making the logic for calculating newWeight not work correctly, and updating it with wrong values

I need to update the entire orderQuery immediately after the first remove mutation. 

I have tried using `queryClient.invalidateQueries({ queryKey: ['Orders'] })` right after the remove-mutation, but it doesnt seem to affect it. 

I've also tried the `setQueriesData` like this:
```
queryClient.setQueryData<OrderModel[]>(['Orders'], (orders) =>
  orders?.map(order =>
    order.orderNo === draft.orderNo && order.lineNo === draft.lineNo
      ? { ...draft }
      : order
  )
);
```

But that doesn't seem to do anything. 
What am I missing here?

Edit: Formatting

**Comment by TkDodo** [comment]
making a mutation doesn't automatically update a query - you need to either invalidate or update directly. I don't know why it doesn't work because I don't really understand that code. Please show a codesandbox reproduction with your issue.
